conda/core/package_cache_data.py

```diff
@@ -729,10 +824,67 @@ def __eq__(self, other):
         return hash(self) == hash(other)
 
 
+# called from ProgressiveFetchExtract.execute()
+def do_cache_action(prec, cache_axn, progress_bar, download_total=1.0):
+    # dummy action to simplify code
+    if not cache_axn:
+        return prec
+    cache_axn.verify()
+
+    if not cache_axn.url.startswith("file:/"):
+
+        def progress_update_cache_axn(pct_completed):
+            progress_bar.update_to(pct_completed * download_total)
+
+    else:
+        download_total = 0
+        progress_update_cache_axn = None
+
+    cache_axn.execute(progress_update_cache_axn)
+    return prec
+
+
+def do_extract_action(prec, extract_axn, progress_bar):
```

Comment:
This function signature leaves me wondering what the arguments are. Is this a place where type hinting could help? Also a doc string here would be very nice 🙏